ibis/expr/visualize.py

```diff
@@ -72,41 +78,44 @@ def get_label(node, argname=None):
 DEFAULT_NODE_ATTRS = {'shape': 'box', 'fontname': 'Deja Vu Sans Mono'}
 
 
-def to_graph(op, node_attr=None, edge_attr=None):
-    if isinstance(op, ir.Expr):
-        return to_graph(op.op())
-    elif not isinstance(op, ops.Node):
-        raise TypeError(op)
+def to_graph(expr, node_attr=None, edge_attr=None):
+    graph = ibis.util.to_op_dag(expr.op())
 
-    stack = [(op, op.name if isinstance(op, ops.Named) else None)]
-    seen = set()
     g = gv.Digraph(
         node_attr=node_attr or DEFAULT_NODE_ATTRS, edge_attr=edge_attr or {}
     )
 
     g.attr(rankdir='BT')
 
-    while stack:
-        op, name = stack.pop()
-
-        if op not in seen:
-            seen.add(op)
-
-            vlabel = get_label(op, argname=name)
-            vhash = str(hash(op))
-            g.node(vhash, label=vlabel)
-
-            for name, arg in zip(op.argnames, op.args):
-                if isinstance(arg, ops.Node):
-                    uhash = str(hash(arg))
-                    ulabel = get_label(arg, argname=name)
-                    g.node(uhash, label=ulabel)
-                    g.edge(uhash, vhash)
-                    stack.append((arg, name))
+    seen = set()
+    edges = set()
+
+    for v, us in graph.items():
+        if isinstance(v, ops.NodeList) and not v:
+            continue
+
+        vhash = str(hash(v))
+        if v not in seen:
+            g.node(vhash, label=get_label(v))
+            seen.add(v)
+
+        for u in us:
+            if isinstance(u, ops.NodeList) and not u:
+                continue
+
+            uhash = str(hash(u))
+            if u not in seen:
+                g.node(uhash, label=get_label(u))
+                seen.add(u)
+            if (edge := (u, v)) not in edges:
+                g.edge(uhash, vhash)
```

Comment:
maybe show arg names as edge lables
```suggestion
                arg_name = None
                if not isinstance(v, ops.NodeList):
                    arg_name = v.argnames[v.args.index(u)]
                g.edge(uhash, vhash, label=arg_name)
```

Reply:
Ah interesting, I was thinking about how to show that but couldn't come up with anything reasonable. Let me see what this looks like!